src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java

```diff
@@ -375,6 +375,16 @@ public IrisSettings saveIrisSettings(ProgrammingExercise exercise, IrisSettings
         }
     }
 
+    /**
+     * Update the Iris settings for a programming exercise.
+     *
+     * @param programmingExerciseBeforeUpdate the programming exercise before the update
+     * @param updatedProgrammingExercise      the programming exercise after the update
+     */
+    public void updateIrisSettings(ProgrammingExercise programmingExerciseBeforeUpdate, ProgrammingExercise updatedProgrammingExercise) {
+        irisSettingsRepository.findByProgrammingExerciseId(programmingExerciseBeforeUpdate.getId()).ifPresent(updatedProgrammingExercise::setIrisSettings);
```

Comment:
Should there be an error or smth when the settings can't be updated (are not present)?

Or: Add this to the documentation so other developers don't have to look into the source code to figure out how this "edge-case" is handled.

Reply:
The settings system will be redone very shortly, so I'll improve it there.